ReactiveCocoa/Swift/Signal.swift

```diff
@@ -125,7 +125,8 @@ public final class Signal<Value, Error: ErrorType> {
 	///
 	/// Returns a Disposable which can be used to disconnect the observer. Disposing
 	/// of the Disposable will have no effect on the Signal itself.
-	public func observe(observer: Observer) -> Disposable? {
+	@discardableResult
```

Comment:
I wonder why you take this stance, since `observe` and its friends have long been having a discardable disposable by not being marked with `@warn_unused_result`. The lifetime is often deterministic when we call `observe` (e.g. `takeUntil`), and so the `Disposable` returned isn't always necessary. I would prefer a cleaner call site instead, unless the result is supposed to be  __always__ used.